Sourcery/Parsing/FileParser.swift

```diff
@@ -344,20 +344,21 @@ extension FileParser {
     fileprivate func parseEnumAssociatedValues(_ body: String) -> [Enum.Case.AssociatedValue] {
         guard !body.isEmpty else { return [] }
 
-        /// name: type, otherType
-        let components = body.components(separatedBy: ",")
-        return components.enumerated().flatMap { idx, element in
-            let nameType = element.components(separatedBy: ":").map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
-
-            switch nameType.count {
-            case 1:
-                return Enum.Case.AssociatedValue(name: "\(idx)", typeName: nameType.first ?? "")
-            case 2:
-                return Enum.Case.AssociatedValue(name: nameType.first, typeName: nameType.last ?? "")
-            default:
-                print("\(logPrefix)parseEnumAssociatedValues: Unknown enum case body format \(body)")
-                return nil
-            }
+        let items = body.commaSeparated()
+        return items
+            .map({ $0.trimmingCharacters(in: .whitespaces) })
+            .enumerated()
+            .map {
+                let nameAndType = $1.colonSeparated().map({ $0.trimmingCharacters(in: .whitespaces) })
+                let defaultName: String? = $0 == 0 && items.count == 1 ? nil : "\($0)"
```

Comment:
yes, it also makes sense to have `indexName` as named elements are also accessible by their index